edge-functions/serverside-analytics/pages/other.tsx

```diff
@@ -0,0 +1,165 @@
+        </Text>
+
+        <ul>
+          <li>
+            <Text variant="description">
+              Home page received {data?.home && data.home} views
+            </Text>
+          </li>
+          <li>
+            <Text variant="description">
+              Other page received {data?.other && data.other} views{' '}
+              {router.asPath === '/other' && '(Current page)'}
+            </Text>
+          </li>
+        </ul>
+        <Text>
+          <Button onClick={router.reload}>Reload Page</Button>
+          <Button className="ml-4" onClick={() => router.push('/  ')}>
+            Switch Page
+          </Button>
+        </Text>
+        <Text>
+          Our Middleware function sends the data in a{' '}
+          <Link href="https://supabase.com/" target="_blank">
+            Supabase
+          </Link>{' '}
+          table. To keep the function light and quick, we are using their rest
+          API endpoint instead of the{' '}
+          <Link href="https://github.com/supabase/supabase-js" target="_blank">
+            SDK
+          </Link>{' '}
+          to avoid the extra loading time to our function.
+        </Text>
+        <Snippet>
+          {`
+import type { NextFetchEvent, NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
+
+async function logPageView(req: NextRequest) {
+  // ignore static assets from being tracked,
+  if (
+    // process.env.NODE_ENV !== 'production' || // uncomment this line to track only production requests
+    req.nextUrl.pathname.startsWith('/api') ||
+    req.nextUrl.pathname.startsWith('/fonts') ||
+    req.nextUrl.pathname.startsWith('/logos') ||
+    req.nextUrl.pathname.startsWith('/images') ||
+    req.nextUrl.pathname.startsWith('/favicon.ico')
+  ) {
+    return
+  }
+
+  //  you can track extra information such as req.geo, req.userAgent, req.headers, etc.
+  const body = JSON.stringify({
+    slug: req.nextUrl.pathname,
+  })
+
+  const request = await fetch(process.env.SUPABASE_URL + "/rest/v1/analytics", {
+    headers: {
+      apikey: process.env.SUPABASE_ANON_KEY!,
+      'Content-Type': 'application/json',
+    },
+    body,
+    method: 'POST',
+  })
+
+  if (request.status !== 201) {
+    console.error('Error logging analytics: ', body)
+  }
+
+  return
+}
+
+export function middleware(req: NextRequest, ev: NextFetchEvent) {
+  const response = NextResponse.next()
+  // Runs after the response has been returned
+  // so tracking analytics doesn't block rendering
+  ev.waitUntil(
+    (async () => {
+      logPageView(req)
+    })()
+  )
+  return response
+}
+
+`}
+          <Text>
+            In order to also capture client-side navigations we can make use of
+            the <Code>routeChangeComplete</Code> event provided by{' '}
+            <Link href="https://nextjs.org/docs/api-reference/next/router#routerevents">
+              <Code>next/router</Code>
+            </Link>
+            .
+          </Text>
+        </Snippet>
```

Comment:
This snippet should close here?

Reply:
That was indeed some wrong pasting on my part!